demos/sample_azure_iot_pnp/sample_azure_iot_pnp.c

```diff
@@ -0,0 +1,1062 @@
+        xResult = AzureIoTJSONWriter_AppendPropertyWithStringValue( xWriter, sampleazureiotCOMMAND_END_TIME,
+                                                                    strlen( sampleazureiotCOMMAND_END_TIME ),
+                                                                    sampleazureiotCOMMAND_FAKE_END_TIME,
+                                                                    strlen( sampleazureiotCOMMAND_FAKE_END_TIME ) );
+    }
+
+    if( xResult == eAzureIoTSuccess )
+    {
+        xResult = AzureIoTJSONWriter_AppendEndObject( xWriter );
+    }
+
+    return xResult;
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Command message callback handler
+ */
+static void prvHandleCommand( AzureIoTHubClientCommandRequest_t * pxMessage,
+                              void * pvContext )
+{
+    AzureIoTResult_t xResult;
+    int32_t lCommandNameLength;
+    int32_t ulCommandPayloadLength;
+    AzureIoTHubClient_t * xHandle = ( AzureIoTHubClient_t * ) pvContext;
+
+    LogInfo( ( "Command payload : %.*s \r\n",
+               pxMessage->ulPayloadLength,
+               pxMessage->pvMessagePayload ) );
+
+    lCommandNameLength = strlen( sampleazureiotCOMMAND_MAX_MIN_REPORT );
+
+    if( ( lCommandNameLength == pxMessage->usCommandNameLength ) &&
+        ( strncmp( sampleazureiotCOMMAND_MAX_MIN_REPORT, pxMessage->pucCommandName, lCommandNameLength ) == 0 ) )
+    {
+        /* Is for max min report */
+        AzureIoTJSONReader_t xReader;
+        xResult = AzureIoTJSONReader_Init( &xReader, pxMessage->pvMessagePayload, pxMessage->ulPayloadLength );
+        configASSERT( xResult == eAzureIoTSuccess );
+
+        AzureIoTJSONWriter_t xWriter;
+        xResult = AzureIoTJSONWriter_Init( &xWriter, ucCommandPayloadBuffer, sizeof( ucCommandPayloadBuffer ) );
+        configASSERT( xResult == eAzureIoTSuccess );
+
+        xResult = prvInvokeMaxMinCommand( &xReader, &xWriter );
+
+        if( xResult == eAzureIoTSuccess )
+        {
+            ulCommandPayloadLength = AzureIoTJSONWriter_GetBytesUsed( &xWriter );
+
+            if( ulCommandPayloadLength > 0 )
+            {
+                if( AzureIoTHubClient_SendCommandResponse( xHandle, pxMessage, 200,
+                                                           ucCommandPayloadBuffer,
+                                                           ulCommandPayloadLength ) != eAzureIoTSuccess )
+                {
+                    LogError( ( "Error sending command response" ) );
+                }
+            }
+        }
+        else
+        {
+            LogError( ( "Error generating command payload" ) );
+
+            if( AzureIoTHubClient_SendCommandResponse( xHandle, pxMessage, 501,
+                                                       sampleazureiotCOMMAND_EMPTY_PAYLOAD,
+                                                       ulCommandPayloadLength ) != eAzureIoTSuccess )
```

Comment:
>ulCommandPayloadLength [](http://example.com/codeflow?start=55&length=22)

ulCommandPayloadLength will have garbage value, use the strlen(sampleazureiotCOMMAND_EMPTY_PAYLOAD)/sizeof(sampleazureiotCOMMAND_EMPTY_PAYLOAD) - 1

Reply:
good catch thank you